webapp/tests/karma/ts/modules/analytics/analytics.component.spec.ts

```diff
@@ -0,0 +1,125 @@
+import { expect } from "chai";
+
+import { AnalyticsComponent } from '@mm-modules/analytics/analytics.component';
+import { AnalyticsModulesService } from '@mm-services/analytics-modules.service';
+import { GlobalActions } from '@mm-actions/global';
+import { AnalyticsActions } from '@mm-actions/analytics';
+
+describe('AnalyticsComponent', () => {
+  let component: AnalyticsComponent;
+  let fixture: ComponentFixture<AnalyticsComponent>;
+  let analyticsModulesService;
+  let globalActions;
+  let analyticsActions;
+  let activatedRoute;
+  let router;
+
+  beforeEach(async(() => {
+    analyticsModulesService = { get: sinon.stub().resolves([]) };
+    const mockSelectors = [
+      { selector: 'getAnalyticsModules', value: [] },
+    ];
+    globalActions = {
+      unsetSelected: sinon.stub(GlobalActions.prototype, 'unsetSelected')
+    };
+    analyticsActions = {
+      setSelectedAnalytics: sinon.stub(AnalyticsActions.prototype, 'setSelectedAnalytics'),
+      setAnalyticsModules: sinon.stub(AnalyticsActions.prototype, 'setAnalyticsModules')
+    };
+    activatedRoute = {
+      snapshot: {
+        routeConfig: { path: 'targets' },
+      },
+    };
+
+    TestBed.configureTestingModule({
+      declarations: [ AnalyticsComponent ],
+      imports: [
+        RouterTestingModule,
+      ],
+      providers: [
+        provideMockStore({ selectors: mockSelectors }),
+        { provide: AnalyticsModulesService, useValue: analyticsModulesService },
+        { provide: ActivatedRoute, useValue: activatedRoute },
+      ]
+    })
+    .compileComponents()
+    .then(() => {
+      fixture = TestBed.createComponent(AnalyticsComponent);
+      component = fixture.componentInstance;
+      analyticsModulesService = TestBed.inject(AnalyticsModulesService);
+      router = TestBed.inject(Router);
+      fixture.detectChanges();
+    });
+  }));
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  it('should create and set up controller with no modules', () => {
+    expect(component).to.exist;
+    expect(component.analyticsModules).to.be.empty;
+    expect(analyticsActions.setSelectedAnalytics.callCount).to.equal(2);
+    expect(globalActions.unsetSelected.callCount).to.equal(1);
+    expect(analyticsModulesService.get.callCount).to.equal(1);
+  });
+
+  it('should set selected the specified module', async () => {
+    sinon.reset();
+    const analyticsModules = [
+      { route: 'reporting' },
+      { route: 'targets' }
+    ];
+    analyticsModulesService.get.resolves([...analyticsModules]);
+
+    await component.getAnalyticsModules();
+
+    expect(analyticsModulesService.get.callCount).to.equal(1);
+    expect(analyticsActions.setSelectedAnalytics.callCount).to.equal(1);
+    expect(analyticsActions.setSelectedAnalytics.args[0][0]).to.equal(analyticsModules[1]);
+    expect(analyticsActions.setAnalyticsModules.callCount).to.equal(1);
+    expect(analyticsActions.setAnalyticsModules.args[0][0]).to.include.members(analyticsModules);
```

Comment:
Are we expecting other analytics modules? Otherwise change to `to.have.members` (which is an unordered array comparison, versus an inclusion).

Reply:
Fixed! Thanks!